### PR TITLE
Handle lack of commitid in languages listing runner

### PR DESCRIPTION
**Pull request by daved**
https://www.pivotaltracker.com/story/show/175567323

 https://www.pivotaltracker.com/story/show/175567323

### CODE REVIEW

File: internal/runners/languages/languages.go [hunk 1]
```diff
@@ -7,38 +7,55 @@ import (
...
+			err, "err_no_usable_commitid", "Cannot obtain a usable commit id",
+		)
+	}
+
+	langs, fail := model.FetchLanguagesForCommit(commitUUID)
+	if fail != nil {
+		return locale.WrapError(
+			fail, "err_fetching_languages",
+			"Cannot obtain languages for commit id {{.V0}}", commitUUID.String(),
+		)
```
Comment (Naatan) [inline internal/runners/languages/languages.go:58]: Why did this need to be added?

**Review by Naatan — CHANGES_REQUESTED** [review]

File: internal/runners/languages/languages.go (hunk 1 above)
Comment (daved) [reply internal/runners/languages/languages.go:58]: In other PRs wrapping at the handler level (i.e the run func) has been emphasized. It also clarifies which commit id was used which errored out.

**Review by Naatan — APPROVED** [review]

File: internal/runners/languages/languages.go [hunk 2]
```diff
@@ -9,36 +9,49 @@ import (
...
 		return locale.NewInputError("err_no_project")
 	}
-	langs, err := model.FetchLanguagesForCommit(l.project.CommitUUID())
-	if err != nil {
-		return err
+
+	commitUUID := l.project.CommitUUID()
+	if commitUUID == "" {
+		return locale.NewError(
+			"err_languages_no_commitid", "A CommitID is required to list languages",
```
Comment (Naatan) [inline internal/runners/languages/languages.go:48]:
```suggestion
			"err_languages_no_commitid", "Your activestate.yaml does not have a commit defined, you may need to run [ACTIONABLE]`state pull`[/RESET] first.",
```

Can we also add a tip here to run `state pull`?

**Review by Naatan — CHANGES_REQUESTED** [review]

File: internal/runners/languages/languages.go [hunk 3]
```diff
@@ -3,42 +3,60 @@ package languages
...
-		return err
+
+	commitUUID := l.project.CommitUUID()
+	if commitUUID == "" {
+		return errs.AddTips(
+			locale.NewError(
+				"err_languages_no_commitid",
+				"Your activestate.yaml does not have a commit defined, you may need to run [ACTIONABLE]`state pull`[/RESET] first.",
+			),
+			locale.Tr("runtime_update_help", l.project.Owner(), l.project.Name()),
```
Comment (Naatan) [inline internal/runners/languages/languages.go:53]: Tips should be short and concise, instantly getting to the point. This one should just say:

```
Run → [ACTIONABLE]state pull[/RESET] to update your project
```

**Review by Naatan — CHANGES_REQUESTED** [review]

**Review by Naatan — APPROVED** [review]

**Pull request merged**